libs/clients/vehicles-mileage/src/lib/vehiclesMileageClient.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Configuration, MileageReadingApi } from '../../gen/fetch'
+import { Provider } from '@nestjs/common'
+import {
+  ConfigType,
+  XRoadConfig,
+  LazyDuringDevScope,
+  IdsClientConfig,
+} from '@island.is/nest/config'
+import { VehiclesMileageClientConfig } from './vehiclesMileageClient.config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+export const VehiclesMileageApiProvider: Provider<MileageReadingApi> = {
+  provide: MileageReadingApi,
+  scope: LazyDuringDevScope,
+  useFactory: (
+    xroadConfig: ConfigType<typeof XRoadConfig>,
+    config: ConfigType<typeof VehiclesMileageClientConfig>,
+    idsClientConfig: ConfigType<typeof IdsClientConfig>,
+  ) =>
+    new MileageReadingApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-mileage-vehicles',
```

Comment:
~ Just a nit pick but just match the library name
```suggestion
          name: 'clients-vehicles-mileage',
```